lib/core/flow.rb

```diff
@@ -17,17 +19,14 @@ def start
     end
   end
 
-  def get_valid_position
-    position = @ui.get_position
-    until @board.grid[position].is_empty? do
-    @ui.show_error_message("Please choose a different position where the cell is empty\n")
-    position = @ui.get_position
-    end
-    position
-  end
-
   def play
-    @board.mark(get_valid_position, @ui.get_symbol)
+    position = @players[0].get_valid_position(@board)
```

Comment:
https://apidock.com/ruby/Array/first ...alternatively, could you make a function that is called current player that does this?